veps/sig-compute/vmtemplate-proposal.md

```diff
@@ -0,0 +1,733 @@
+KubeVirt, but also the import of appliances created by software vendors and
+templates created through other tools. As a widely adopted open standard in the
+virtualization industry, OVA/OVF is the ideal format for KubeVirt to implement.
+
+The entry point for the import mechanism should be a new
+`VirtualMachineTemplateImportRequest `CRD, which contains the required fields to
+specify a remote location where an OVA/OVF file can be obtained from and where
+it should be imported to. Another option could be the direct upload into the
+cluster through `virtctl` and possibly with the help of CDI.
+
+Once the OVA/OVF file can be processed, the contained disks need to be imported
+into PVCs and the attached virtual machine definition needs to be converted into
+a KubeVirt `VirtualMachine`. Once these tasks are completed the controller can
+create a new `VirtualMachineTemplate` from the results. This new template can
+then be used to create actual `VirtualMachines` from the imported template.
+
+Similar to the import mechanism, the entry point to the export mechanism should
+be a new `VirtualMachineTemplateExportRequest` CRD containing the required
+fields to specify either an existing `VirtualMachineTemplate` to be exported or
+an existing `VirtualMachine` to be exported directly. Once the exported template
+is ready to be downloaded, the controller should make the URL to download the
+template available in the status of the new CRD.
+
+For both import and export, `virtctl` should be extended to enable the feature
+to be used easily via the CLI.
+
+## API Examples
+
+For API examples, see the code blocks in the `template.kubevirt.io/v1alpha1`
+design.
+
+## Alternatives
+
+### Develop feature inside the core KubeVirt repository
+
+The current design proposes to develop this feature outside the KubeVirt core
+repository. An alternative would be to develop the feature within the KubeVirt
+core repository. However, this would lead to a slowdown in development as it
+would further increase the burden on reviewers in the core repository, would
+require all KubeVirt e2e tests to be run, most of which have nothing to do with
+the feature, and it would further increase the already substantial codebase.
+
+### Use existing k8s workload template solutions such as Helm or Kustomize
+
+As discussed within the Motivation section, there are many drawbacks to using
+existing out of cluster k8s template tooling such as Helm and Kustomize to
+template VMs. Most notably, these solutions do not manage the underlying storage
+state, a critical requirement for templating stateful workloads like VMs.
+
+### Extend the containerdisks to include a VirtualMachine definition
+
+This could be an alternative to using a standardized format such as OVA/OVF to
+import and export VM templates. However, this would be a proprietary solution
+that is only supported by KubeVirt and does not allow interoperability with the
+larger virtualization ecosystem.
+
```

Comment:
What about introducing a containertemplate format that encapsulate an OVA/OVF?

That would make sharing easier and more container native.